### DISCUSSION TITLE
part-ii/2004-5/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#70] [Question comments]
Part II 2004 Graph Theory: A4.9 

https://questions.tripos.org/part-ii/2004-5/

**Comment by walrus1729** [comment]
<details>
<summary>Solutions to a), b), c), d)</summary>

a) True. Indeed, if $G$ is acyclic, then $e(G) \geq n - 1$. To see this, we note that if we add an edge within a connected component of a graph, then this creates a cycle. So if we construct an acyclic graph by adding edges to an empty graph on $n$ vertices, then each edge must connect two different components and therefore reduce the number of components of the graph by 1. But we start with $n$ components, so we can only do this $n-1$ times. Hence if $G$ is acyclic with $e(G) \leq n - 1$ (so in fact $e(G) = n - 1$), then $G$ is maximal acyclic, so it is a tree.

b) True. If $G$ is connected, then $e(G) \geq n - 1$, since the empty graph has $n$ components and adding an edge decreases the number of components by at most 1. Hence if $G$ is connected with $e(G) \leq n - 1$ (so in fact $e(G) = n - 1$), then $G$ is minimal connected, so it is a tree.

c) False. A counterexample is a 4-cycle with two of its vertices also each connected to another vertex (different for the two vertices); this has adjacency matrix
$\begin{pmatrix}
0 & 1 & 0 & 0 & 0 & 0 \\
1 & 0 & 1 & 1 & 0 & 0 \\
0 & 1 & 0 & 0 & 1 & 0 \\
0 & 1 & 0 & 0 & 1 & 0 \\
0 & 0 & 1 & 1 & 0 & 1 \\
0 & 0 & 0 & 0 & 1 & 0 \\
\end{pmatrix}$

d) False. A counterexample is a path of length 5 (which is a tree), along with a graph whose components are a path of length 2 and a 3-cycle. Both have degree sequence 1, 1, 2, 2, 2. The former has adjacency matrix 
$\begin{pmatrix}
0 & 1 & 0 & 0 & 0 \\
1 & 0 & 1 & 0 & 0 \\
0 & 1 & 0 & 1 & 0 \\
0 & 0 & 1 & 0 & 1 \\
0 & 0 & 0 & 1 & 0 \\
\end{pmatrix}$
and the latter has adjacency matrix
$\begin{pmatrix}
0 & 1 & 0 & 0 & 0 \\
1 & 0 & 0 & 0 & 0 \\
0 & 0 & 0 & 1 & 1 \\
0 & 0 & 1 & 0 & 1 \\
0 & 0 & 1 & 1 & 0 \\
\end{pmatrix}$

</details>
